playbooks/maas-pre-flight.yml

```diff
@@ -49,6 +49,158 @@
     - maas-pre-flight
 
 
+- name: Dynamic variable generation
+  hosts: localhost
+  connection: local
+  gather_facts: false
+  become: true
+  pre_tasks:
+    - name: Check for /etc/openstack-release
+      stat:
+        path: /etc/openstack-release
+      register: _osa_release_file
+
```

Comment:
from line 62-74 I think we can wrap all that in a block making it conditional the existence of the release file.

``` yaml
- name: osa release block
  block:
    - name: Register /etc/openstack-release version
      shell: grep DISTRIB_RELEASE /etc/openstack-release | tr -dc '0-9.'
      register: _maas_product_osa_release_version

    - name: Set OSA release version fact
      set_fact:
        maas_product_osa_release_version: "{{ _maas_product_osa_release_version.stdout.split('.')[0] }}"
  when:
    - not deploy_osp | bool
    - _osa_release_file.stat.exists
```